generator/lib/src/generator.dart

```diff
@@ -297,8 +302,50 @@ class RetrofitGenerator extends GeneratorForAnnotation<http.RestApi> {
 
     return _getResponseInnerType(generic);
   }
+
+  void _generateExtra(
+      MethodElement m, List<Code> blocks, String localExtraVar) {
+    final extra =
+        _typeChecker(dio.Extra).firstAnnotationOf(m, throwOnUnresolved: false);
+
+    if (extra != null) {
+      final c = ConstantReader(extra);
+      blocks.add(literalMap(
+        c.peek('data')?.mapValue?.map((k, v) {
+              return MapEntry(
+                k.toBoolValue() ??
+                    k.toDoubleValue() ??
+                    k.toIntValue() ??
+                    k.toStringValue() ??
+                    k.toListValue() ??
+                    k.toMapValue() ??
+                    k.toSetValue() ??
+                    k.toSymbolValue() ??
+                    k.toTypeValue(),
+                v.toBoolValue() ??
+                    v.toDoubleValue() ??
+                    v.toIntValue() ??
+                    v.toStringValue() ??
+                    v.toListValue() ??
+                    v.toMapValue() ??
+                    v.toSetValue() ??
+                    v.toSymbolValue() ??
+                    v.toTypeValue(),
+              );
+            }) ??
+            {},
+        refer('String'),
+        refer('dynamic'),
+      ).assignConst(localExtraVar).statement);
+    } else {
+      blocks.add(literalMap(
+        {},
+        refer('String'),
+        refer('dynamic'),
+      ).assignConst(localExtraVar).statement);
+    }
+  }
 }
 
 Builder generatorFactoryBuilder({String header}) =>
-    new PartBuilder([new RetrofitGenerator()], ".retrofit.dart",
```

Comment:
Any special reason to change `.retrofit.dart` to `.g.dart`?

Reply:
I like to have all generated files should have `.g.dart` extension.
If user has multiple builders, they all should produce generated output in a  shared `.g.dart` file. This makes the code very manageable and understandable. What if I have many apis and many models. Each of them will have a separate file `.retrofit.dart` for RestClient and `.g.dart` for models.

If a builder produces `file_name.retrofit.dart` it is not clear that `file_name.retrofit.dart` is a generated file. But any new developer to the project will understand `file_name.g.dart` is a generated file.

Another benefit is that we are using `SharedPartBuilder` instead of `PartBuilder`, so if the user wants to see the generated code by each builder, They can easily see different code generated by different builders in the `.g.dart` file
because `SharedPartBuilder` automatically adds comment before each generator. So it's very clear which builder generated which output.

Example:
```dart
part of 'my_file.dart'

// **************************************************************************
// RetrofitGenerator
// **************************************************************************

/// Here will be code by `retrofit`


// **************************************************************************
// SomeOtherGenerator
// **************************************************************************

/// Here will be code by `some_other_builder`
```


There are 2 generated parts. Which is unnecessary, since both are generated.

```dart
import 'package:some_other_builder/some_other_builder.dart';
import 'package:retrofit/retrofit.dart';

/// Generated by `some_other_builder`
part 'my_file.g.dart';

/// Generated by `retrofit`
part 'my_file.retrofit.dart';
```


From dart-lang/build/blob/master/docs/writing_a_builder.md#builders-which-produce-an-unknown-number-of-outputs

> Dart does not have restrictions like Java does - we don't need a 1:1 mapping between files and classes.

So we don't need to have separate files for each builder. This is not required in dart.

But still if you don't want it. Tell me I will change it back to `.retrofit.dart`

Reply:
It's OK.